src/AutoUI/Filters/Filters.tsx

```diff
@@ -58,22 +43,27 @@ const removeFieldsWithNoFilter = (schema: JSONSchema): JSONSchema => {
 				newProperties[key] = value;
 				continue;
 			}
+			// Apply removal logic in case our parent has defined an array x-no-filter for its children
+			if (parentXNoFilterSet?.has(key)) {
```

Comment:
I was doing it to make it works at any levels

Reply:
Can it though?
If it's meant to work many levels deep, wouldn't it need to provide the full nested path to avoid conflicts, following the same format as the `x-ref-scheme`?
For example if the application model on its device prop says `x-no-filter: ['name']`, would it mean the name of the device env vars or the device config vars?

Reply:
btw I don't see us using `string[]` for `x-no-filter` at all in the UI atm, so I think it make sense to start stricter and make it looser later.

Reply:
@thgreasi as you rightly got, my previous version was totally broken on that part, I forgot to change the filter when I switched from a set to an array... 

so my point was to make the following in the filter function, to work in a bland manner :

x-no-filter: ["slug", "is_of__cpu_architecture[0].slug"]

['is_of__cpu_architecture[0].slug'].some(string => string.split('.').includes('slug'))

which is not correct and would not work for all cases, but was something to start with. The best solution would be to do what I was doing with generateSchemaFromRefScheme (changing the function) which accept the property and construct it removing the nested property described if necessary.


for sure your solution is improving mine, as mine was totally broken at that point.